sideboard/config.py

```diff
@@ -9,14 +10,72 @@
 from validate import Validator
 
 
+_defaults_re = re.compile(r'(.+)-defaults(\.\w+)$')
+
+
+def uniquify(x):
+    """
+    Returns an ordered copy of `x` with duplicate items removed.
+
+    >>> uniquify(['a', 'b', 'a', 'c', 'a', 'd', 'a', 'e'])
+    ['a', 'b', 'c', 'd', 'e']
+
+    """
+    seen = set()
+    return [i for i in x if i not in seen and not seen.add(i)]
```

Comment:
Two suggestions about this function:

1) You should probably call ``listify`` on the parameter just to be safe.  I'm a big fan if ``listify`` :)

2) Purely a style suggestion: I like the Haskell convention of using ``xs`` as the variable name for "a generic iterable" and ``x`` as a variable name for "a generic element".  We use that convention elsewhere in Sideboard, and if you used it here then we'd say ``[x for x in listify(xs) if x not in seen and not seen.add(x)]``.  The main reason why I like that more than ``for i in x`` is that ``i`` is usually the variable name used for indexes, and I did a double take when first reading this function to make sure we were collecting elements rather than indexes.